### PR TITLE
KOA-5726 BPKOverlay

### PR DESCRIPTION
**Pull request by gert-janvercauteren**
# BPKOverlay

## Solid

| Day | Night |
| --- | --- | 
| iPhone-overlay___solid_dm | iPhone-overlay___solid_lm |

## Top

| Day | Night |
| --- | --- | 
| iPhone-overlay___top_lm | iPhone-overlay___top_dm |

## Bottom

| Day | Night |
| --- | --- | 
| iPhone-overlay___bottom_lm | iPhone-overlay___bottom_dm |

## Left

| Day | Night |
| --- | --- | 
| iPhone-overlay___left_lm | iPhone-overlay___left_dm |

## Right

| Day | Night |
| --- | --- | 
| iPhone-overlay___right_lm | iPhone-overlay___right_dm |

## Vignette

| Day | Night |
| --- | --- | 
| iPhone-overlay___vignette_lm | iPhone-overlay___vignette_dm |

+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] `README.md`
+ [ ] Tests
+ [ ] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [ ] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/OverlayView/Classes/BPKOverlayView.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+public class BPKOverlayView: UIView {
+    public enum OverlayType {
```
Comment (frugoman) [inline Backpack/OverlayView/Classes/BPKOverlayView.swift:22]: can we drop the `Overlay` prefix? its already inside the overlay so its redundant I think

File: Backpack/Overlay/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+| Day | Night |
+| --- | --- |
+| <img src="https://raw.githubusercontent.com/Skyscanner/backpack-ios/main/screenshots/iPhone-overlay___left_lm.png" alt="" width="375" /> |<img src="https://raw.githubusercontent.com/Skyscanner/backpack-ios/main/screenshots/iPhone-overlay___left_dm.png" alt="" width="375" /> |
+
+## Right
+
+| Day | Night |
+| --- | --- |
+| <img src="https://raw.githubusercontent.com/Skyscanner/backpack-ios/main/screenshots/iPhone-overlay___right_lm.png" alt="" width="375" /> |<img src="https://raw.githubusercontent.com/Skyscanner/backpack-ios/main/screenshots/iPhone-overlay___right_dm.png" alt="" width="375" /> |
```
[the remark this answers is not in the document]
Comment (gert-janvercauteren) [reply Backpack/Overlay/README.md:35]: Sharp! Updated the README

File: Backpack/Overlay/Classes/VignetteOverlay.swift [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        let right = shadowRadius + shadowOffset.width
+        let shadowRect = CGRect(
+            x: bounds.origin.x - left,
+            y: bounds.origin.y - top,
+            width: bounds.width + left + right,
+            height: bounds.height + top + bottom
+        )
+        
+        let path = CGMutablePath()
+        let delta: CGFloat = 1
```
Comment (frugoman) [inline Backpack/Overlay/Classes/VignetteOverlay.swift:90]: :chefskiss:

**Review by frugoman — APPROVED** [review]
Awesome work! 🚢 🚀

**Pull request merged**